infra/gcp/namespaces/ensure-namespaces.sh

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
+#
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script creates namespaces, applies roles and role bindings and
+# checks if assigned role binding is the only one in the namespace
+# for k8s-infra projects
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_DIR=$(dirname "${BASH_SOURCE[0]}")
+. "${SCRIPT_DIR}/lib.sh"
+
+role_filename="namespace-user-role.yml"
+role_binding_filename="namespace-user-role-binding.yml"
+role_binding_name_postfix="-namespace-user-binding"
+
+function usage() {
+    echo "usage: $0" > /dev/stderr
+    echo > /dev/stderr
+}
+
+if [ $# != 0 ]; then
+    usage
+    exit 1
+fi
+
+function create_namespace() {
+  kubectl create "$1"
+}
+
+function apply_role() {
+  role=$(sed -e "s/{{namespace}}/$1/" "$role_filename")
+
+  kubectl apply -f "$role"
+}
+
+function apply_role_binding() {
+  role_binding=$(sed -e "s/{{namespace}}/$1/" "$role_binding_filename")
+
+  kubectl apply -f "$role_binding"
+}
+
+function ensure_only_one_proper_role_binding() {
+  project_name="$1"
+  role_bindings_count="$(kubectl get rolebindings -o json -n "$project_name" | jq -r '.items | length')"
+
+  if [ "$role_bindings_count" -gt 1 ]; then
+    echo "Only one role binding per namespace is allowed (current: $role_bindings_count)"
+    exit 1
+  fi
+
+  role_binding_name="$(kubectl get rolebindings -o json -n "$project_name" | jq -r '.items[0].metadata.name')"
+
+  if [ "$role_binding_name" != "${project_name}${role_binding_name_postfix}" ]; then
```

Comment:
The object names no longer has the project_name in it, right?  So you don't need postfix, just "namespace-user" literal

Reply:
Fixed